x/fee_grant/types/expiration.go

```diff
@@ -8,35 +8,43 @@ import (
 
 // ExpiresAtTime creates an expiration at the given time
 func ExpiresAtTime(t time.Time) ExpiresAt {
-	return ExpiresAt{Time: t}
+	return ExpiresAt{
+		Sum: &ExpiresAt_Time{
+			Time: &t,
+		},
+	}
 }
 
 // ExpiresAtHeight creates an expiration at the given height
 func ExpiresAtHeight(h int64) ExpiresAt {
-	return ExpiresAt{Height: h}
+	return ExpiresAt{
+		&ExpiresAt_Height{
+			Height: h,
+		},
+	}
 }
 
 // ValidateBasic performs basic sanity checks.
 // Note that empty expiration is allowed
 func (e ExpiresAt) ValidateBasic() error {
-	if !e.Time.IsZero() && e.Height != 0 {
+	if e.GetTime() != nil && !e.GetTime().IsZero() && e.GetHeight() != 0 {
 		return sdkerrors.Wrap(ErrInvalidDuration, "both time and height are set")
 	}
-	if e.Height < 0 {
+	if e.GetHeight() < 0 {
 		return sdkerrors.Wrap(ErrInvalidDuration, "negative height")
 	}
 	return nil
 }
 
 // IsZero returns true for an uninitialized struct
 func (e ExpiresAt) IsZero() bool {
```

Comment:
How about renaming  this function to `Undefined`? 
`IsZero` for this use-case is hacky.